statsmodels/stats/proportion.py

```diff
@@ -569,6 +569,9 @@ def proportions_ztest(count, nobs, value=None, alternative='two-sided',
     TODO: add continuity correction or other improvements for small samples.
 
     '''
+    count=np.asarray(count)
+    nobs=np.asarray(nobs)
```

Comment:
You can add this to this PR.  Yes, if nobs is not scalar and not the same shape as count, then we should raise a ValueError. 

You could also check the unit tests to see how much of this is covered.

full vectorization will require more work, and checking to see how it is supposed to be interpreted, and would be too much for this PR